kotlinx.fuzz.gradle/src/test/kotlin/kotlinx.fuzz.gradle.junit.test/AnnotationsTest.kt

```diff
@@ -0,0 +1,82 @@
+package kotlinx.fuzz.gradle.junit.test
+
+import kotlin.reflect.KFunction
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.fuzz.ConfigurationException
+import kotlinx.fuzz.KFuzzTest
+import kotlinx.fuzz.KFuzzer
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Disabled
+import org.junit.jupiter.api.Test
+import org.junit.platform.engine.TestExecutionResult
+import org.junit.platform.engine.discovery.DiscoverySelectors.selectMethod
+import org.junit.platform.testkit.engine.EngineTestKit
+
+class AnnotationsTest {
+    @BeforeEach
+    fun setup() {
+        writeToSystemProperties {
+            maxSingleTargetFuzzTime = 5.seconds
+            instrument = listOf("kotlinx.fuzz.test.**")
+            workDir = kotlin.io.path.createTempDirectory("fuzz-test")
+        }
+    }
+
+    @KFuzzTest(maxFuzzTime = "not even a duration")
+    fun invalidConfig(data: KFuzzer) {
+        error("unreachable")
+    }
+
+    @KFuzzTest(keepGoing = 2)
+    fun overriddenConfig(data: KFuzzer) {
+        // A _hacky_ way to test if config params are actually overridden.
+        // It's based on probability of fuzzer finding the second exception before first being basically 0.
+        // If keepGoing=2 is not set, we will only get the first error.
+        if (data.boolean()) {
+            error("first error")
+        }
+        if (data.int() * data.int() == 25) {
+            error("second error")
+        }
+    }
+
+    @Test
+    fun testInvalidConfig() {
+        val result = runMethodFuzz(AnnotationsTest::invalidConfig)
+        assertEquals(TestExecutionResult.Status.FAILED, result.status)
+        val exception = result.throwable!!.get().cause!!  // parent exception is from junit
+        assertIs<ConfigurationException>(
+            exception,
+            message = "wrong exception type\n${exception.stackTraceToString()}",
+        )
+    }
+
+    // TODO: enable once keepGoing works
+    @Disabled
+    @Test
+    fun testOverriddenConfig() {
+        val result = runMethodFuzz(AnnotationsTest::overriddenConfig)
+        assertEquals(TestExecutionResult.Status.FAILED, result.status)
+        val exception = result.throwable!!.get()  // no parent exception! TODO: should there be?
```

Comment:
It shouldn't be here, because the test started successfully

Reply:
I don't actually see reasons to wrap an additional layer of exceptions here. It will affect only what the user sees and therefore it is better to have distinctive and representative exceptions immediately, imho